proto/teleport/lib/teleterm/v1/tshd_events_service.proto

```diff
@@ -146,6 +162,55 @@ message PromptMFAResponse {
   string totp_code = 1;
 }
 
+// Request for PromptHardwareKeyPINAsk.
+message PromptHardwareKeyPINAskRequest {
+  string root_cluster_uri = 1;
+  // Message to display in the prompt.
+  string message = 2;
```

Comment:
That's why I don't like passing arbitrary messages through this RPC. 😩 The square brackets make sense in the context of CLI prompts, with the square brackets meaning "the default thing that will happen when you press enter without providing any input". In the context of a GUI, it makes little sense and looks awkward.

<img width="1280" alt="square-brackets" src="https://github.com/user-attachments/assets/fff73105-ec2b-4de0-9aca-16341074d791">

I feel like the proper way to solve it would be to make it so that instead of passing a string in `y.prompt.AskPIN(ctx, "Enter your YubiKey PIV PIN")`, we pass an enum value. Each prompt implementation can map the enum value to whatever it wants. Otherwise our GUIs will keep suffering from surfacing CLI-specific messages.

I don't think this warrants blocking this PR, maybe a followup if you find time for it?

Reply:
Agree, this looks poor, I guess I didn't have enough motivation to change this 😶‍🌫️
It's not too much to improve it however, so I simply changed it in this PR (otherwise I'd have to reserve the `message` field in the RPC later, it's better to avoid it).

I added the enum on `HardwareKeyPrompt` interface level, do you think it makes sense to have an enum in the RPC too? (I went with a bool there).

Reply:
Ah, so the way I was imagining this is that every possible message would be a separate enum value. But looking at the changes now, I guess it's not strictly necessary.

> I added the enum on `HardwareKeyPrompt` interface level, do you think it makes sense to have an enum in the RPC too? (I went with a bool there).

I'd say in general it's better to use enums all the way through so that it's easier to add values to them. Chances are that if an enum needs to be extended, the next person is going to add an enum value and then just enough boolean field. But it's fine for now, especially since that bool is just in the internal API.